### PR TITLE
Add caching

### LINKED ISSUE #4: Caching layers?
Hi, will you implement caching via GitHub of the layers created by Nixpacks?

### PR DESCRIPTION
**Pull request by PolpOnline**
This pull request introduces new caching functionality for the build process. 
Currently tested on [My repo](https://github.com/PolpOnline/monitor).
Note that the platform feature will probably not work with cache on.

closes #4

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -32,6 +32,13 @@ inputs:
     description: "Boolean to determine if the built image should be pushed"
     required: false
     default: 'false'
+  cache:
+    description: "Boolean to determine if the build should use the cache"
+    required: false
+    default: 'false'
+  cache_tag:
```
Comment (iloveitaly) [inline action.yml:39]: can we select smart defaults here?

can we note it's required when `cache=true`?

File: README.md [hunk 2]
```diff
@@ -20,6 +20,9 @@ It's very opinionated out the box (as software should be!) but allows you to cus
 - `pkgs`: Optional additional Nix packages to install in the environment.
 - `apt`: Optional additional Apt packages to install in the environment.
 - `push`: A boolean flag to indicate whether to push the built image to the registry. Default is `false`. Required for multi-architecture builds.
+- `cache`: A boolean flag to indicate whether to use the build cache. Default is `false`. 
+  (NOTE: The cache is shared between all builds in the repository. Some cache metadata will be inlined in the final image.)
+- `cache_tag`: A single tag to use for the cache image. Required if `cache` is `true`.
```
Comment (iloveitaly) [inline README.md:28]: Can you give an example?

File: README.md [hunk 3]
```diff
@@ -20,6 +20,9 @@ It's very opinionated out the box (as software should be!) but allows you to cus
 - `pkgs`: Optional additional Nix packages to install in the environment.
 - `apt`: Optional additional Apt packages to install in the environment.
 - `push`: A boolean flag to indicate whether to push the built image to the registry. Default is `false`. Required for multi-architecture builds.
+- `cache`: A boolean flag to indicate whether to use the build cache. Default is `false`.
```
Comment (iloveitaly) [inline README.md]: can you note why cache is important / link to nixpacks docs on caching?

**Review by iloveitaly — CHANGES_REQUESTED** [review]
Looks great! I'd like to add smart defaults here, could you add those in?

File: README.md (hunk 2 above)
Comment (PolpOnline) [reply README.md:28]: I added a default value note, should be sufficient as an example

**Comment by iloveitaly** [comment]
Looks great, thanks. 

Is there a good reason not to enable caching by default?

**Comment by PolpOnline** [comment]
@iloveitaly My reason is that a user might get distracted and enable caching with platforms also enables, that might cause some issues

**Pull request merged**